compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml

```diff
@@ -598,8 +598,9 @@ roundCommercial d x =
 zero: NumericScale n => Numeric n
 zero = intToNumeric 0
 
-half: NumericScale n => Numeric n
-half = fromNumeric (0.5: Numeric 1)
+-- | fail for scale 0
+half: forall n. NumericScale n => Numeric n
+half = primitive @"BECastNumeric" (numericOne @n) (0.5: Numeric 1)
```

Comment:
@akrmn what do you think of the change ? 
e.g. `half @0` was `0.` before now it fails.

Reply:
> `half @0` was `0.` before

I don't think this is true, on `2.9.0-snapshot.20231220.0` (random SDK I had around) `half @0` gives me an `ArithmeticError` so it was already failing. I think it's fine for it to fail, this also fails with the same error:

```
fromNumeric @(Numeric 1) (0.25 : Numeric 2)
```

Regardless, I don't have a strong preference between this and the previous version, this one inlines the primitive so it might be slightly more efficient, but the previous one is easier to understand. I do prefer the explicitly quantified `n` but I'll leave it to you

Reply:
It is fail, I will inline the result of the previous version. 

Thanks for double checking